hlx_statics/blocks/discoverblock/discoverblock.js

```diff
@@ -1,81 +1,83 @@
-      const parentElement = block.parentElement;
-      parentElement.appendChild(existingWrapper);
-    }
-    existingWrapper.appendChild(block);
-    const firstHeading = existingWrapper.querySelector('h2, h3');
-    if (firstHeading) {
-      Array.from(existingWrapper.children).forEach((child,index) => {
-        if(index > 0) {
-          child.style.marginTop = '35px';
-        } 
-      });
+  // Set width based on data-width attribute
+  const width = block.getAttribute('data-width');
+  const wrapper = block.closest('.discoverblock-wrapper');
+
+  // If there's an image, set the block width to 2*1280/12 and position image on the left
+  const hasImage = block.querySelector('img') !== null;
+  if (hasImage) {
+    let imageWidth = (2 * 1280) / 12;
+    imageWidth = imageWidth + 156;
+    wrapper.style.width = imageWidth + 'px';
+
+    // Get the image
+    const image = block.querySelector('img');
+
+    // Set image width to 100px and move it to the beginning
+    image.style.width = '100px';
+    image.style.height = 'auto';
+    image.style.flexShrink = '0';
+    image.style.position = 'absolute';
+
+    // Move the image to be the first child (leftmost position)
+    block.insertBefore(image, block.firstChild);
+
+    // Find button-container's parent and give it a class name
+    const buttonContainer = block.querySelector('.button-container');
+    if (buttonContainer && buttonContainer.parentElement) {
+      buttonContainer.parentElement.classList.add('discover-content-with-image');
     }
-    const children = Array.from(existingWrapper.children);
-    children.forEach(child => {
-      child.style.maxWidth = '290px';
-    });
-    const discoverBlocks = document.querySelectorAll('.discoverblock-wrapper');
-    discoverBlocks.forEach(div => {
-      if (div.childElementCount === 0) {
-        div.remove();
-      }
-    });
+  } else if (width) {
+    // if data-width exists, override the default width with the data-width.
+    wrapper.style.width = width;
   }
-  
-  Array.from(block.children).forEach(div => {
-    const containsHeading = div.querySelector('h1, h2, h3, h4, h5, h6') !== null;
-    if (containsHeading) {
-      var breakDiv = createTag('div', { class: 'break' });
 
-      block.insertBefore(breakDiv, div);
-    }
-    else {
-      div.classList.add('discover-nonheading-child-container')
+  const heading = block.querySelector('h1, h2, h3, h4, h5, h6');
+  if (heading) {
+    // Check if block has image and add class to heading if it does
+    if (hasImage) {
+      heading.classList.add('discover-heading-with-image');
     }
+    const headingClone = heading.cloneNode(true);
+
+    // Add anchor link if it's h2 or h3
+    if (headingClone.tagName === 'H2' || headingClone.tagName === 'H3') {
+      const headingText = headingClone.textContent.trim();
+      const anchorId = headingText.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/^-+|-+$/g, '');
```

Comment:
We can just use headingId, similar to what decorateHeading is doing in https://github.com/AdobeDocs/adp-devsite/blob/e6000d352e9bf007338a968c83588479be9bc42a/hlx_statics/components/heading.js#L23-L24
```suggestion
      const anchorId = heading.id;
```